BatteryChecker.h

```diff
@@ -0,0 +1,56 @@
+#define TEMP_MIN			0
+#define TEMP_MAX			45
+#define SOC_MIN				20
+#define SOC_MAX				80
+#define CHARGERATE_MIN		0
+#define CHARGERATE_MAX  	0.8
+#define TOLERANCE			5/100
+
+typedef enum{
+	TEMPERATURE,
+	SOC,
+	CHARGERATE
+}ten_BatteryParameter;
+
+typedef enum{
+	CELCIUS,
+	FAHRENHEIT
+}ten_TempUnit;
+
+typedef enum{

```

Comment:
There are three enums here, one for each parameter type. Addition of a new level (e.g., pre-warning) will result in multiple additions. Similarly, addition of a new parameter will result in repeating all levels in another enum.
Think single-responsibility for data as well. Try separating the levels as one enum and the parameters as another.